### ISSUE TITLE
Pixi error due to version number

### ISSUE THREAD

**Issue by traversaro** [conda-forge/yarp-feedstock#42]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

> DEPRECATION: yarp 3.9.0~dev has a non-standard version number. pip 24.1 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of yarp or contact the author to suggest that they release a version with a conforming version number. Discussion can be found at https://github.com/pypa/pip/issues/12063

We should just drop the `~dev` .

### Installed packages

```shell
(icub-models-generator) traversaro@IITBMP014LW012:~/icub-models-generator$ pixi list
Package                               Version       Build                  Size       Kind   Source
_libgcc_mutex                         0.1           conda_forge            2.5 KiB    conda  _libgcc_mutex-0.1-conda_forge.tar.bz2
_openmp_mutex                         4.5           2_gnu                  23.1 KiB   conda  _openmp_mutex-4.5-2_gnu.tar.bz2
ace                                   7.1.3         h59595ed_2             5.2 MiB    conda  ace-7.1.3-h59595ed_2.conda
alsa-lib                              1.2.10        hd590300_0             541.9 KiB  conda  alsa-lib-1.2.10-hd590300_0.conda
ampl-mp                               3.1.0         h2cc385e_1006          1.1 MiB    conda  ampl-mp-3.1.0-h2cc385e_1006.tar.bz2
aom                                   3.8.1         h59595ed_0             2.6 MiB    conda  aom-3.8.1-h59595ed_0.conda
assimp                                5.3.1         hfb0e8fe_2             3.4 MiB    conda  assimp-5.3.1-hfb0e8fe_2.conda
attr                                  2.5.1         h166bdaf_1             69.4 KiB   conda  attr-2.5.1-h166bdaf_1.tar.bz2
bzip2                                 1.0.8         hd590300_5             248.3 KiB  conda  bzip2-1.0.8-hd590300_5.conda
c-ares                                1.26.0        hd590300_0             158.9 KiB  conda  c-ares-1.26.0-hd590300_0.conda
ca-certificates                       2024.2.2      hbcca054_0             151.8 KiB  conda  ca-certificates-2024.2.2-hbcca054_0.conda
cairo                                 1.18.0        h3faef2a_0             959.3 KiB  conda  cairo-1.18.0-h3faef2a_0.conda
catkin_pkg                            1.0.0         pyhd8ed1ab_0           52.2 KiB   conda  catkin_pkg-1.0.0-pyhd8ed1ab_0.conda
dav1d                                 1.2.1         hd590300_0             742.4 KiB  conda  dav1d-1.2.1-hd590300_0.conda
dbus                                  1.13.6        h5008d03_3             604.1 KiB  conda  dbus-1.13.6-h5008d03_3.tar.bz2
docutils                              0.20.1        py312h7900ff3_3        877.2 KiB  conda  docutils-0.20.1-py312h7900ff3_3.conda
eigen                                 3.4.0         h00ab1b0_0             1 MiB      conda  eigen-3.4.0-h00ab1b0_0.conda
expat                                 2.5.0         hcb278e6_1             133.6 KiB  conda  expat-2.5.0-hcb278e6_1.conda
ffmpeg                                6.1.1         gpl_h8007c5b_104       9.3 MiB    conda  ffmpeg-6.1.1-gpl_h8007c5b_104.conda
font-ttf-dejavu-sans-mono             2.37          hab24e00_0             388.1 KiB  conda  font-ttf-dejavu-sans-mono-2.37-hab24e00_0.tar.bz2
font-ttf-inconsolata                  3.000         h77eed37_0             94.3 KiB   conda  font-ttf-inconsolata-3.000-h77eed37_0.tar.bz2
font-ttf-source-code-pro              2.038         h77eed37_0             684.4 KiB  conda  font-ttf-source-code-pro-2.038-h77eed37_0.tar.bz2
font-ttf-ubuntu                       0.83          h77eed37_1             1.5 MiB    conda  font-ttf-ubuntu-0.83-h77eed37_1.conda
fontconfig                            2.14.2        h14ed4e7_0             265.6 KiB  conda  fontconfig-2.14.2-h14ed4e7_0.conda
fonts-conda-ecosystem                 1             0                      3.6 KiB    conda  fonts-conda-ecosystem-1-0.tar.bz2
fonts-conda-forge                     1             0                      4 KiB      conda  fonts-conda-forge-1-0.tar.bz2
freeglut                              3.2.2         hac7e632_2             139.6 KiB  conda  freeglut-3.2.2-hac7e632_2.conda
freetype                              2.12.1        h267a509_2             620.1 KiB  conda  freetype-2.12.1-h267a509_2.conda
fribidi                               1.0.10        h36c2ea0_0             111.7 KiB  conda  fribidi-1.0.10-h36c2ea0_0.tar.bz2
gettext                               0.21.1        h27087fc_0             4.1 MiB    conda  gettext-0.21.1-h27087fc_0.tar.bz2
glfw                                  3.3.9         hd590300_0             125.2 KiB  conda  glfw-3.3.9-hd590300_0.conda
glib                                  2.78.3        hfc55251_0             478.1 KiB  conda  glib-2.78.3-hfc55251_0.conda
glib-tools                            2.78.3        hfc55251_0             108.1 KiB  conda  glib-tools-2.78.3-hfc55251_0.conda
gmp                                   6.3.0         h59595ed_0             549.9 KiB  conda  gmp-6.3.0-h59595ed_0.conda
gnutls                                3.7.9         hb077bed_0             1.9 MiB    conda  gnutls-3.7.9-hb077bed_0.conda
graphite2                             1.3.13        h58526e2_1001          102.2 KiB  conda  graphite2-1.3.13-h58526e2_1001.tar.bz2
gst-plugins-base                      1.22.9        h8e1006c_0             2.6 MiB    conda  gst-plugins-base-1.22.9-h8e1006c_0.conda
gstreamer                             1.22.9        h98fc4e7_0             1.9 MiB    conda  gstreamer-1.22.9-h98fc4e7_0.conda
harfbuzz                              8.3.0         h3d44ed6_0             1.5 MiB    conda  harfbuzz-8.3.0-h3d44ed6_0.conda
hdf5                                  1.14.3        nompi_h4f84152_100     3.7 MiB    conda  hdf5-1.14.3-nompi_h4f84152_100.conda
icu                                   73.2          h59595ed_0             11.5 MiB   conda  icu-73.2-h59595ed_0.conda
idyntree                              10.3.0        py312h36ac6ef_0        2.3 MiB    conda  idyntree-10.3.0-py312h36ac6ef_0.conda
ipopt                                 3.14.14       h04b96a2_1             996.7 KiB  conda  ipopt-3.14.14-h04b96a2_1.conda
irrlicht                              1.8.5         h2a6caf8_4             1.9 MiB    conda  irrlicht-1.8.5-h2a6caf8_4.conda
jasper                                4.2.0         he6dfbbe_0             660.5 KiB  conda  jasper-4.2.0-he6dfbbe_0.conda
keyutils                              1.6.1         h166bdaf_0             115.1 KiB  conda  keyutils-1.6.1-h166bdaf_0.tar.bz2
krb5                                  1.21.2        h659d440_0             1.3 MiB    conda  krb5-1.21.2-h659d440_0.conda
lame                                  3.100         h166bdaf_1003          496.3 KiB  conda  lame-3.100-h166bdaf_1003.tar.bz2
ld_impl_linux-64                      2.40          h41732ed_0             688.2 KiB  conda  ld_impl_linux-64-2.40-h41732ed_0.conda
lerc                                  4.0.0         h27087fc_0             275.2 KiB  conda  lerc-4.0.0-h27087fc_0.tar.bz2
libabseil                             20230802.1    cxx17_h59595ed_0       1.2 MiB    conda  libabseil-20230802.1-cxx17_h59595ed_0.conda
libaec                                1.1.2         h59595ed_1             34.4 KiB   conda  libaec-1.1.2-h59595ed_1.conda
libass                                0.17.1        h8fe9dca_1             123.9 KiB  conda  libass-0.17.1-h8fe9dca_1.conda
libblas                               3.9.0         21_linux64_openblas    14.3 KiB   conda  libblas-3.9.0-21_linux64_openblas.conda
libboost                              1.82.0        h6fcfa73_6             2.5 MiB    conda  libboost-1.82.0-h6fcfa73_6.conda
libcap                                2.69          h0f662aa_0             98.2 KiB   conda  libcap-2.69-h0f662aa_0.conda
libcblas                              3.9.0         21_linux64_openblas    14.3 KiB   conda  libcblas-3.9.0-21_linux64_openblas.conda
libclang                              15.0.7        default_hb11cfb5_4     130.3 KiB  conda  libclang-15.0.7-default_hb11cfb5_4.conda
libclang13                            15.0.7        default_ha2b6cf4_4     9.1 MiB    conda  libclang13-15.0.7-default_ha2b6cf4_4.conda
libcups                               2.3.3         h4637d8d_4             4.3 MiB    conda  libcups-2.3.3-h4637d8d_4.conda
libcurl                               8.5.0         hca28451_0             380 KiB    conda  libcurl-8.5.0-hca28451_0.conda
libdeflate                            1.19          hd590300_0             65.5 KiB   conda  libdeflate-1.19-hd590300_0.conda
libdrm                                2.4.114       h166bdaf_0             298 KiB    conda  libdrm-2.4.114-h166bdaf_0.tar.bz2
libedit                               3.1.20191231  he28a2e2_2             121 KiB    conda  libedit-3.1.20191231-he28a2e2_2.tar.bz2
libev                                 4.33          hd590300_2             110.1 KiB  conda  libev-4.33-hd590300_2.conda
libevent                              2.1.12        hf998b51_1             417.4 KiB  conda  libevent-2.1.12-hf998b51_1.conda
libexpat                              2.5.0         hcb278e6_1             76.2 KiB   conda  libexpat-2.5.0-hcb278e6_1.conda
libffi                                3.4.2         h7f98852_5             56.9 KiB   conda  libffi-3.4.2-h7f98852_5.tar.bz2
libflac                               1.4.3         h59595ed_0             385.1 KiB  conda  libflac-1.4.3-h59595ed_0.conda
libgcc-ng                             13.2.0        h807b86a_5             752.4 KiB  conda  libgcc-ng-13.2.0-h807b86a_5.conda
libgcrypt                             1.10.3        hd590300_0             620 KiB    conda  libgcrypt-1.10.3-hd590300_0.conda
libgfortran-ng                        13.2.0        h69a702a_5             23.3 KiB   conda  libgfortran-ng-13.2.0-h69a702a_5.conda
libgfortran5                          13.2.0        ha4646dd_5             1.4 MiB    conda  libgfortran5-13.2.0-ha4646dd_5.conda
libglib                               2.78.3        h783c2da_0             2.6 MiB    conda  libglib-2.78.3-h783c2da_0.conda
libglu                                9.0.0         hac7e632_1003          323.5 KiB  conda  libglu-9.0.0-hac7e632_1003.conda
libgomp                               13.2.0        h807b86a_5             409.9 KiB  conda  libgomp-13.2.0-h807b86a_5.conda
libgpg-error                          1.47          h71f35ed_0             254.7 KiB  conda  libgpg-error-1.47-h71f35ed_0.conda
libhwloc                              2.9.3         default_h554bfaf_1009  2.5 MiB    conda  libhwloc-2.9.3-default_h554bfaf_1009.conda
libi2c                                4.3           hcb278e6_2             18.2 KiB   conda  libi2c-4.3-hcb278e6_2.conda
libiconv                              1.17          hd590300_2             689.2 KiB  conda  libiconv-1.17-hd590300_2.conda
libidn2                               2.3.7         hd590300_0             123.5 KiB  conda  libidn2-2.3.7-hd590300_0.conda
libjpeg-turbo                         3.0.0         hd590300_1             604.1 KiB  conda  libjpeg-turbo-3.0.0-hd590300_1.conda
liblapack                             3.9.0         21_linux64_openblas    14.3 KiB   conda  liblapack-3.9.0-21_linux64_openblas.conda
liblapacke                            3.9.0         21_linux64_openblas    14.3 KiB   conda  liblapacke-3.9.0-21_linux64_openblas.conda
libllvm15                             15.0.7        hb3ce162_4             31.8 MiB   conda  libllvm15-15.0.7-hb3ce162_4.conda
libnghttp2                            1.58.0        h47da74e_1             617.1 KiB  conda  libnghttp2-1.58.0-h47da74e_1.conda
libnsl                                2.0.1         hd590300_0             32.6 KiB   conda  libnsl-2.0.1-hd590300_0.conda
libogg                                1.3.4         h7f98852_1             205.6 KiB  conda  libogg-1.3.4-h7f98852_1.tar.bz2
libopenblas                           0.3.26        pthreads_h413a1c8_0    5.3 MiB    conda  libopenblas-0.3.26-pthreads_h413a1c8_0.conda
libopencv                             4.9.0         py311haea74c2_7        29.4 MiB   conda  libopencv-4.9.0-py311haea74c2_7.conda
libopenvino                           2023.3.0      h2e90f83_0             5.7 MiB    conda  libopenvino-2023.3.0-h2e90f83_0.conda
libopenvino-auto-batch-plugin         2023.3.0      hd5fc58b_0             112.6 KiB  conda  libopenvino-auto-batch-plugin-2023.3.0-hd5fc58b_0.conda
libopenvino-auto-plugin               2023.3.0      hd5fc58b_0             233 KiB    conda  libopenvino-auto-plugin-2023.3.0-hd5fc58b_0.conda
libopenvino-hetero-plugin             2023.3.0      h3ecfda7_0             177.5 KiB  conda  libopenvino-hetero-plugin-2023.3.0-h3ecfda7_0.conda
libopenvino-intel-cpu-plugin          2023.3.0      h2e90f83_0             9.7 MiB    conda  libopenvino-intel-cpu-plugin-2023.3.0-h2e90f83_0.conda
libopenvino-intel-gpu-plugin          2023.3.0      h2e90f83_0             7.8 MiB    conda  libopenvino-intel-gpu-plugin-2023.3.0-h2e90f83_0.conda
libopenvino-ir-frontend               2023.3.0      h3ecfda7_0             194.3 KiB  conda  libopenvino-ir-frontend-2023.3.0-h3ecfda7_0.conda
libopenvino-onnx-frontend             2023.3.0      hfbc7f12_0             1.5 MiB    conda  libopenvino-onnx-frontend-2023.3.0-hfbc7f12_0.conda
libopenvino-paddle-frontend           2023.3.0      hfbc7f12_0             644.2 KiB  conda  libopenvino-paddle-frontend-2023.3.0-hfbc7f12_0.conda
libopenvino-pytorch-frontend          2023.3.0      h59595ed_0             937.9 KiB  conda  libopenvino-pytorch-frontend-2023.3.0-h59595ed_0.conda
libopenvino-tensorflow-frontend       2023.3.0      h0bff32c_0             1.1 MiB    conda  libopenvino-tensorflow-frontend-2023.3.0-h0bff32c_0.conda
libopenvino-tensorflow-lite-frontend  2023.3.0      h59595ed_0             445.7 KiB  conda  libopenvino-tensorflow-lite-frontend-2023.3.0-h59595ed_0.conda
libopus                               1.3.1         h7f98852_1             254.5 KiB  conda  libopus-1.3.1-h7f98852_1.tar.bz2
libosqp                               0.6.3         h59595ed_0             70.4 KiB   conda  libosqp-0.6.3-h59595ed_0.conda
libpciaccess                          0.17          h166bdaf_0             38.8 KiB   conda  libpciaccess-0.17-h166bdaf_0.tar.bz2
libpng                                1.6.42        h2797004_0             282.3 KiB  conda  libpng-1.6.42-h2797004_0.conda
libpq                                 16.2          h33b98f1_0             2.4 MiB    conda  libpq-16.2-h33b98f1_0.conda
libprotobuf                           4.25.1        hf27288f_1             2.6 MiB    conda  libprotobuf-4.25.1-hf27288f_1.conda
libqdldl                              0.1.5         h27087fc_1             16.5 KiB   conda  libqdldl-0.1.5-h27087fc_1.tar.bz2
libscotch                             7.0.4         h91e35bf_1             331.7 KiB  conda  libscotch-7.0.4-h91e35bf_1.conda
libsndfile                            1.2.2         hc60ed4a_1             346.1 KiB  conda  libsndfile-1.2.2-hc60ed4a_1.conda
libspral                              2023.09.07    h6aa6db2_2             277.8 KiB  conda  libspral-2023.09.07-h6aa6db2_2.conda
libsqlite                             3.45.1        h2797004_0             839.2 KiB  conda  libsqlite-3.45.1-h2797004_0.conda
libssh2                               1.11.0        h0841786_0             264.8 KiB  conda  libssh2-1.11.0-h0841786_0.conda
libstdcxx-ng                          13.2.0        h7e041cc_5             3.7 MiB    conda  libstdcxx-ng-13.2.0-h7e041cc_5.conda
libsystemd0                           255           h3516f8a_0             394.8 KiB  conda  libsystemd0-255-h3516f8a_0.conda
libtasn1                              4.19.0        h166bdaf_0             114.1 KiB  conda  libtasn1-4.19.0-h166bdaf_0.tar.bz2
libtiff                               4.6.0         ha9c0a0a_2             276.6 KiB  conda  libtiff-4.6.0-ha9c0a0a_2.conda
libunistring                          0.9.10        h7f98852_0             1.4 MiB    conda  libunistring-0.9.10-h7f98852_0.tar.bz2
libuuid                               2.38.1        h0b41bf4_0             32.8 KiB   conda  libuuid-2.38.1-h0b41bf4_0.conda
libva                                 2.20.0        hd590300_0             183.7 KiB  conda  libva-2.20.0-hd590300_0.conda
libvorbis                             1.3.7         h9c3ff4c_0             279.6 KiB  conda  libvorbis-1.3.7-h9c3ff4c_0.tar.bz2
libvpx                                1.13.1        h59595ed_0             982.5 KiB  conda  libvpx-1.13.1-h59595ed_0.conda
libwebp-base                          1.3.2         hd590300_0             392.4 KiB  conda  libwebp-base-1.3.2-hd590300_0.conda
libxcb                                1.15          h0b41bf4_0             375.2 KiB  conda  libxcb-1.15-h0b41bf4_0.conda
libxcrypt                             4.4.36        hd590300_1             98 KiB     conda  libxcrypt-4.4.36-hd590300_1.conda
libxkbcommon                          1.6.0         hd429924_1             561.4 KiB  conda  libxkbcommon-1.6.0-hd429924_1.conda
libxml2                               2.12.5        h232c23b_0             688.3 KiB  conda  libxml2-2.12.5-h232c23b_0.conda
libxslt                               1.1.39        h76b75d6_0             248.3 KiB  conda  libxslt-1.1.39-h76b75d6_0.conda
libyarp                               3.9.0         ha614a09_1             9.7 MiB    conda  libyarp-3.9.0-ha614a09_1.conda
libzlib                               1.2.13        hd590300_5             60.1 KiB   conda  libzlib-1.2.13-hd590300_5.conda
lxml                                  4.9.4         py312h37b5203_0        1.4 MiB    conda  lxml-4.9.4-py312h37b5203_0.conda
lz4-c                                 1.9.4         hcb278e6_0             140 KiB    conda  lz4-c-1.9.4-hcb278e6_0.conda
metis                                 5.1.0         h59595ed_1007          3.7 MiB    conda  metis-5.1.0-h59595ed_1007.conda
mpg123                                1.32.4        h59595ed_0             479.6 KiB  conda  mpg123-1.32.4-h59595ed_0.conda
mumps-include                         5.6.2         ha770c72_4             26 KiB     conda  mumps-include-5.6.2-ha770c72_4.conda
mumps-seq                             5.6.2         hfef103a_4             1.9 MiB    conda  mumps-seq-5.6.2-hfef103a_4.conda
mysql-common                          8.0.33        hf1915f5_6             735.8 KiB  conda  mysql-common-8.0.33-hf1915f5_6.conda
mysql-libs                            8.0.33        hca2cd23_6             1.5 MiB    conda  mysql-libs-8.0.33-hca2cd23_6.conda
ncurses                               6.4           h59595ed_2             863.7 KiB  conda  ncurses-6.4-h59595ed_2.conda
nettle                                3.9.1         h7ab15ed_0             987.9 KiB  conda  nettle-3.9.1-h7ab15ed_0.conda
nspr                                  4.35          h27087fc_0             221.5 KiB  conda  nspr-4.35-h27087fc_0.conda
nss                                   3.97          h1d7d5a4_0             1.9 MiB    conda  nss-3.97-h1d7d5a4_0.conda
numpy                                 1.26.4        py312heda63a1_0        7.1 MiB    conda  numpy-1.26.4-py312heda63a1_0.conda
ocl-icd                               2.3.2         hd590300_0             133.1 KiB  conda  ocl-icd-2.3.2-hd590300_0.conda
ocl-icd-system                        1.0.0         1                      4.2 KiB    conda  ocl-icd-system-1.0.0-1.tar.bz2
openh264                              2.4.1         h59595ed_0             718 KiB    conda  openh264-2.4.1-h59595ed_0.conda
openssl                               3.2.1         hd590300_0             2.7 MiB    conda  openssl-3.2.1-hd590300_0.conda
osqp-eigen                            0.8.1         hdd734ac_0             34.9 KiB   conda  osqp-eigen-0.8.1-hdd734ac_0.conda
p11-kit                               0.24.1        hc5aa10d_0             4.5 MiB    conda  p11-kit-0.24.1-hc5aa10d_0.tar.bz2
pcre2                                 10.42         hcad00b1_0             993.4 KiB  conda  pcre2-10.42-hcad00b1_0.conda
pip                                   24.0          pyhd8ed1ab_0           1.3 MiB    conda  pip-24.0-pyhd8ed1ab_0.conda
pixman                                0.43.2        h59595ed_0             377.8 KiB  conda  pixman-0.43.2-h59595ed_0.conda
portaudio                             19.6.0        h7c63dc7_9             112.8 KiB  conda  portaudio-19.6.0-h7c63dc7_9.conda
pthread-stubs                         0.4           h36c2ea0_1001          5.5 KiB    conda  pthread-stubs-0.4-h36c2ea0_1001.tar.bz2
pugixml                               1.14          h59595ed_0             112.2 KiB  conda  pugixml-1.14-h59595ed_0.conda
pulseaudio-client                     16.1          hb77b528_5             737.2 KiB  conda  pulseaudio-client-16.1-hb77b528_5.conda
pyparsing                             3.1.1         pyhd8ed1ab_0           87.4 KiB   conda  pyparsing-3.1.1-pyhd8ed1ab_0.conda
python                                3.12.1        hab00c5b_1_cpython     30.8 MiB   conda  python-3.12.1-hab00c5b_1_cpython.conda
python-dateutil                       2.8.2         pyhd8ed1ab_0           240.2 KiB  conda  python-dateutil-2.8.2-pyhd8ed1ab_0.tar.bz2
python_abi                            3.12          4_cp312                6.2 KiB    conda  python_abi-3.12-4_cp312.conda
pyyaml                                6.0.1         py312h98912ed_1        192 KiB    conda  pyyaml-6.0.1-py312h98912ed_1.conda
qt-main                               5.15.8        h5810be5_19            58.5 MiB   conda  qt-main-5.15.8-h5810be5_19.conda
readline                              8.2           h8228510_1             274.9 KiB  conda  readline-8.2-h8228510_1.conda
robot-testing-framework               2.0.1         hcb278e6_1             181.1 KiB  conda  robot-testing-framework-2.0.1-hcb278e6_1.conda
scotch                                7.0.4         h23d43cc_1             89.9 KiB   conda  scotch-7.0.4-h23d43cc_1.conda
sdl                                   1.2.68        h293081c_0             152.5 KiB  conda  sdl-1.2.68-h293081c_0.conda
sdl2                                  2.28.5        h77f46ba_0             1.3 MiB    conda  sdl2-2.28.5-h77f46ba_0.conda
setuptools                            69.0.3        pyhd8ed1ab_0           459.5 KiB  conda  setuptools-69.0.3-pyhd8ed1ab_0.conda
six                                   1.16.0        pyh6c4a22f_0           13.9 KiB   conda  six-1.16.0-pyh6c4a22f_0.tar.bz2
snappy                                1.1.10        h9fff704_0             38 KiB     conda  snappy-1.1.10-h9fff704_0.conda
soxr                                  0.1.3         h0b41bf4_3             128.3 KiB  conda  soxr-0.1.3-h0b41bf4_3.conda
svt-av1                               1.8.0         h59595ed_0             2.5 MiB    conda  svt-av1-1.8.0-h59595ed_0.conda
tbb                                   2021.11.0     h00ab1b0_1             191 KiB    conda  tbb-2021.11.0-h00ab1b0_1.conda
tinyxml                               2.6.2         h4bd325d_2             55.2 KiB   conda  tinyxml-2.6.2-h4bd325d_2.tar.bz2
tk                                    8.6.13        noxft_h4845f30_101     3.2 MiB    conda  tk-8.6.13-noxft_h4845f30_101.conda
tzdata                                2024a         h0c530f3_0             117 KiB    conda  tzdata-2024a-h0c530f3_0.conda
unixodbc                              2.3.12        h661eb56_0             275.2 KiB  conda  unixodbc-2.3.12-h661eb56_0.conda
wheel                                 0.42.0        pyhd8ed1ab_0           56.2 KiB   conda  wheel-0.42.0-pyhd8ed1ab_0.conda
x264                                  1!164.3095    h166bdaf_2             876.5 KiB  conda  x264-1!164.3095-h166bdaf_2.tar.bz2
x265                                  3.5           h924138e_3             3.2 MiB    conda  x265-3.5-h924138e_3.tar.bz2
xcb-util                              0.4.0         hd590300_1             19.3 KiB   conda  xcb-util-0.4.0-hd590300_1.conda
xcb-util-image                        0.4.0         h8ee46fc_1             23.9 KiB   conda  xcb-util-image-0.4.0-h8ee46fc_1.conda
xcb-util-keysyms                      0.4.0         h8ee46fc_1             13.9 KiB   conda  xcb-util-keysyms-0.4.0-h8ee46fc_1.conda
xcb-util-renderutil                   0.3.9         hd590300_1             16.6 KiB   conda  xcb-util-renderutil-0.3.9-hd590300_1.conda
xcb-util-wm                           0.4.1         h8ee46fc_1             50.9 KiB   conda  xcb-util-wm-0.4.1-h8ee46fc_1.conda
xkeyboard-config                      2.41          hd590300_0             877 KiB    conda  xkeyboard-config-2.41-hd590300_0.conda
xorg-fixesproto                       5.0           h7f98852_1002          8.9 KiB    conda  xorg-fixesproto-5.0-h7f98852_1002.tar.bz2
xorg-inputproto                       2.3.2         h7f98852_1002          19.1 KiB   conda  xorg-inputproto-2.3.2-h7f98852_1002.tar.bz2
xorg-kbproto                          1.0.7         h7f98852_1002          26.7 KiB   conda  xorg-kbproto-1.0.7-h7f98852_1002.tar.bz2
xorg-libice                           1.1.1         hd590300_0             57.1 KiB   conda  xorg-libice-1.1.1-hd590300_0.conda
xorg-libsm                            1.2.4         h7391055_0             26.8 KiB   conda  xorg-libsm-1.2.4-h7391055_0.conda
xorg-libx11                           1.8.7         h8ee46fc_0             809.3 KiB  conda  xorg-libx11-1.8.7-h8ee46fc_0.conda
xorg-libxau                           1.0.11        hd590300_0             14.1 KiB   conda  xorg-libxau-1.0.11-hd590300_0.conda
xorg-libxcursor                       1.2.0         h0b41bf4_1             30.6 KiB   conda  xorg-libxcursor-1.2.0-h0b41bf4_1.conda
xorg-libxdmcp                         1.1.3         h7f98852_0             18.7 KiB   conda  xorg-libxdmcp-1.1.3-h7f98852_0.tar.bz2
xorg-libxext                          1.3.4         h0b41bf4_2             49 KiB     conda  xorg-libxext-1.3.4-h0b41bf4_2.conda
xorg-libxfixes                        5.0.3         h7f98852_1004          17.7 KiB   conda  xorg-libxfixes-5.0.3-h7f98852_1004.tar.bz2
xorg-libxi                            1.7.10        h7f98852_0             46.2 KiB   conda  xorg-libxi-1.7.10-h7f98852_0.tar.bz2
xorg-libxinerama                      1.1.5         h27087fc_0             13 KiB     conda  xorg-libxinerama-1.1.5-h27087fc_0.tar.bz2
xorg-libxrandr                        1.5.2         h7f98852_1             29 KiB     conda  xorg-libxrandr-1.5.2-h7f98852_1.tar.bz2
xorg-libxrender                       0.9.11        hd590300_0             36.9 KiB   conda  xorg-libxrender-0.9.11-hd590300_0.conda
xorg-randrproto                       1.5.0         h7f98852_1001          32.2 KiB   conda  xorg-randrproto-1.5.0-h7f98852_1001.tar.bz2
xorg-renderproto                      0.11.1        h7f98852_1002          9.4 KiB    conda  xorg-renderproto-0.11.1-h7f98852_1002.tar.bz2
xorg-xextproto                        7.3.0         h0b41bf4_1003          29.6 KiB   conda  xorg-xextproto-7.3.0-h0b41bf4_1003.conda
xorg-xf86vidmodeproto                 2.3.1         h7f98852_1002          23.3 KiB   conda  xorg-xf86vidmodeproto-2.3.1-h7f98852_1002.tar.bz2
xorg-xproto                           7.0.31        h7f98852_1007          73.2 KiB   conda  xorg-xproto-7.0.31-h7f98852_1007.tar.bz2
xz                                    5.2.6         h166bdaf_0             408.6 KiB  conda  xz-5.2.6-h166bdaf_0.tar.bz2
yaml                                  0.2.5         h7f98852_2             87.1 KiB   conda  yaml-0.2.5-h7f98852_2.tar.bz2
yarp                                  3.9.0         ha770c72_1             22.6 KiB   conda  yarp-3.9.0-ha770c72_1.conda
yarp-python                           3.9.0         py312hebe80a8_1        1.1 MiB    conda  yarp-python-3.9.0-py312hebe80a8_1.conda
ycm-cmake-modules                     0.16.2        h59595ed_0             138 KiB    conda  ycm-cmake-modules-0.16.2-h59595ed_0.conda
zlib                                  1.2.13        hd590300_5             90.6 KiB   conda  zlib-1.2.13-hd590300_5.conda
zstd                                  1.5.5         hfc55251_0             532.4 KiB  conda  zstd-1.5.5-hfc55251_0.conda
```

### Environment info

```shell
(icub-models-generator) traversaro@IITBMP014LW012:~/icub-models-generator$ pixi info
      Pixi version: 0.13.0
          Platform: linux-64
  Virtual packages: __unix=0=0
                  : __linux=5.15.133.1=0
                  : __glibc=2.35=0
                  : __cuda=12.3=0
                  : __archspec=1=x86_64
         Cache dir: /home/traversaro/.cache/rattler/cache
      Auth storage: /home/traversaro/.rattler/credentials.json

Project
------------
           Version: 0.0.0
     Manifest file: /home/traversaro/icub-models-generator/pixi.toml
      Last updated: 15-02-2024 02:36:39

Environments
------------
default
          Features: default
          Channels: conda-forge
  Dependency count: 8
      Dependencies: idyntree, yarp, lxml, pyyaml, numpy, catkin_pkg, pip, python
  Target platforms: linux-64, osx-arm64, win-64, linux-aarch64, osx-64
```

**Comment by traversaro** [comment]
Latest pixi started to raise an error:
~~~
  /home/runner/.pixi/bin/pixi install --locked --manifest-path pixi.toml --color always
  thread 'main' panicked at /home/runner/work/pixi/pixi/src/install_pypi.rs:385:58:
  could not create site-packages: Failed to read metadata: from /home/runner/work/icub-models-generator/icub-models-generator/.pixi/envs/default/lib/python3.11/site-packages/yarp-3.9.0~dev.dist-info
  
  Caused by:
      after parsing 3.9.0, found "~dev" after it, which is not part of a valid version
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
~~~

See https://github.com/robotology/icub-models-generator/actions/runs/8373685322/job/22927273151, we should go ahead and fix this.

**Comment by traversaro** [comment]
> Latest pixi started to raise an error:
> 
> ```
>   /home/runner/.pixi/bin/pixi install --locked --manifest-path pixi.toml --color always
>   thread 'main' panicked at /home/runner/work/pixi/pixi/src/install_pypi.rs:385:58:
>   could not create site-packages: Failed to read metadata: from /home/runner/work/icub-models-generator/icub-models-generator/.pixi/envs/default/lib/python3.11/site-packages/yarp-3.9.0~dev.dist-info
>   
>   Caused by:
>       after parsing 3.9.0, found "~dev" after it, which is not part of a valid version
>   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> See https://github.com/robotology/icub-models-generator/actions/runs/8373685322/job/22927273151, we should go ahead and fix this.

fyi @Nicogene @martinaxgloria I guess you may encounter this in icub-tests. This should be solved in https://github.com/conda-forge/yarp-feedstock/pull/43, but unless you refresh the lock file, you will still get the old yarp version. In that case probably you can use pixi 0.15 and everything should work fine.

**Comment by traversaro** [comment]
Let's see how it goes with https://github.com/prefix-dev/pixi/issues/1031 . If that is not fixed in pixi, probably it could make sense to backport this fix at least for yarp 3.8.1 .

**Comment by traversaro** [comment]
> Let's see how it goes with [prefix-dev/pixi#1031](https://github.com/prefix-dev/pixi/issues/1031) . If that is not fixed in pixi, probably it could make sense to backport this fix at least for yarp 3.8.1 .

In the end I did not do that. If at least a user complains, we can do that.
